app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt

```diff
@@ -323,10 +320,20 @@ class FormView constructor(
         imm?.hideSoftInputFromWindow(binding.recyclerView.windowToken, 0)
     }
 
-    private fun doesItemNeedsKeyboard(item: FieldUiModel) = when (item) {
```

Comment:
ValueType enum has some utility methods like `isText` which you may use here instead of having such a big list

Reply:
Very good tip!, thanks.